backend/module/chaos/serverexperimentation/rtds/rtds_cache.go

```diff
@@ -86,7 +155,7 @@ func refreshCache(ctx context.Context, store experimentstore.ExperimentStore, sn
 	}
 }
 
-func setSnapshot(snapshotCache gcpCache.SnapshotCache, rtdsLayerName string, upstreamCluster string,
+func setSnapshot(snapshotCache cacheWrapper, rtdsLayerName string, upstreamCluster string,
```

Comment:
I really like the enhancements in this function (along with the necessary change). Thank you